cli/popper/gha.py

```diff
@@ -406,71 +407,78 @@ def run(self, reuse):
             pu.fail('Action {} failed!\n'.format(self.action['name']))
 
     def docker_exists(self):
-        cmd_out, _ = pu.exec_cmd('docker ps -a',
-                                 debug=self.debug, dry_run=self.dry_run)
+        if self.dry_run:
+            return True
+        container = self.docker_client.containers.list(
+            all=True, filters={'name': self.cid})
 
-        if self.cid in cmd_out:
+        container = [c for c in container if c.name == self.cid]
+        if container:
+            self.container = container[0]
             return True
 
         return False
 
     def docker_rm(self):
-        pu.exec_cmd('docker rm {}'.format(self.cid),
-                    debug=self.debug, dry_run=self.dry_run)
+        if self.dry_run:
+            return
+        self.container.remove(force=True)
 
     def docker_create(self, img):
+        pu.info('{}[{}] docker create {} {}\n'.format(
+            self.msg_prefix,
+            self.action['name'], img, ' '.join(self.action.get('args', ''))
+        ))
+        if self.dry_run:
+            return
         env_vars = self.action.get('env', {})
 
         for s in self.action.get('secrets', []):
             env_vars.update({s: os.environ[s]})
 
         for e, v in self.env.items():
             env_vars.update({e: v})
-
         env_vars.update({'HOME': os.environ['HOME']})
+        volumes = [self.workspace, os.environ['HOME'], '/var/run/docker.sock']
+        self.container = self.docker_client.containers.create(
+            image=img, command=self.action.get('args', None),
+            name=self.cid, volumes={v: {'bind': v} for v in volumes},
+            working_dir=self.workspace, environment=env_vars,
+            entrypoint=self.action.get('runs', None))
```

Comment:
when `self.debug` is `True`, we can show a message saying that we're about to invoke `create()`. And also another one below when we're executing `start()`